platforms/s32k1xx/bsp/bspUart/src/Uart.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright 2024 Accenture.
+// Copyright 2025 BMW AG
+
+#include "bsp/UartConfig.h"
+
+#include <bsp/clock/boardClock.h>
+
+using bsp::Uart;
+
+static uint32_t const WRITE_TIMEOUT = 1000U;
+
+void Uart::init()
+{
+    (void)bios::Io::setDefaultConfiguration(_uartConfig.txPin);
+    (void)bios::Io::setDefaultConfiguration(_uartConfig.rxPin);
+    _uartConfig.uart.GLOBAL = 0;
+    _uartConfig.uart.CTRL   = 0;
+
+    _uartConfig.uart.PINCFG = 0;
+    _uartConfig.uart.BAUD   = 0;
+    _uartConfig.uart.BAUD   = _uartConfig.baudRate[0].baud;
+    _uartConfig.uart.STAT   = 0xFFFFFFFFU;
+    _uartConfig.uart.STAT   = 0;
+    _uartConfig.uart.MODIR  = 0;
+
+    _uartConfig.uart.FIFO  = 0;
+    _uartConfig.uart.WATER = 0;
+    // Last
+    _uartConfig.uart.CTRL  = LPUART_CTRL_RE(1U) + LPUART_CTRL_TE(1U);
+}
+
+bool Uart::isRxReady() const
```

Comment:
Could this be a non member static function taking `LPUART_Type&`? If yes it should.